internal/host/host.go

```diff
@@ -1264,7 +1264,7 @@ func (m *Manager) selectRole(ctx context.Context, h *models.Host, db *gorm.DB) (
 		return autoSelectedRole, err
 	}
 
-	if len(masters) < common.MinMasterHostsNeededForInstallation {
+	if len(masters) < common.MinMasterHostsNeededForInstallationInHaMode {
```

Comment:
I've started to test this change for [AGENT-989](https://issues.redhat.com/browse/AGENT-989) and the role assignment appeared to not work correctly for ABI. If I set the control plane replicas to 5 (in ABI would be in agent-cluster-install). 3 nodes would be set to master, and remaining 2 nodes would be assigned workers. 

I think the issue is the number of control plane replicas isn't flowing from from [AgentClusterInstall to CreateClusterParams](https://github.com/openshift/assisted-service/blob/818d92f978db0f0bbfa8e911affdaf31c02dd177/cmd/agentbasedinstaller/register.go#L90).

Rather it is used to [determine HighAvailabilityMode](https://github.com/openshift/assisted-service/blob/818d92f978db0f0bbfa8e911affdaf31c02dd177/internal/controller/controllers/clusterdeployments_controller.go#L1318). There isn't a field in CreateClusterParams to set the number control plane replicas. 

In order to install a cluster with 5 control plane nodes, should ABI be setting the hosts roles to master before starting cluster installation?

Reply:
Commented on https://redhat-internal.slack.com/archives/C02SPBZ4GPR/p1730014829899899?thread_ts=1729626990.388869&cid=C02SPBZ4GPR

Reply:
Hi @danmanor. I tested your latest updates using ABI and setting 5 controlplane agents in AgentClusterInstall still result in 3 masters and 2 workers. I think CreateClusterParams needs to be updated, please see my comment here: https://redhat-internal.slack.com/archives/C02SPBZ4GPR/p1730231838967219?thread_ts=1729626990.388869&cid=C02SPBZ4GPR

Reply:
Thanks for the update Richard. It shouldn't relate to `CreateClusterParams` as it is used for cluster registration but not for host registration. I will investigate further and update